contracts/randdrop/src/contract.rs

```diff
@@ -0,0 +1,763 @@
+            execute_register_merkle_root(deps, env, info, merkle_root)
+        }
+        //RandDrop should be called by the manager with a future timestamp
+        ExecuteMsg::Randdrop { amount, proof } => execute_randdrop(deps, info, amount, proof),
+        //NoisReceive should be called by the proxy contract. The proxy is forwarding the randomness from the nois chain to this contract.
+        ExecuteMsg::NoisReceive { callback } => execute_receive(deps, env, info, callback),
+        ExecuteMsg::Claim {} => execute_claim(deps, env, info),
+        ExecuteMsg::WithdrawAll { address } => execute_withdraw_all(deps, env, info, address),
+    }
+}
+
+#[entry_point]
+pub fn query(deps: Deps, _env: Env, msg: QueryMsg) -> StdResult<QueryResponse> {
+    let response = match msg {
+        QueryMsg::IsWinner { address } => to_binary(&query_is_winner(deps, address)?)?,
+        QueryMsg::Config {} => to_binary(&query_config(deps)?)?,
+        QueryMsg::MerkleRoot {} => to_binary(&query_merkle_root(deps)?)?,
+        QueryMsg::IsClaimed { address } => to_binary(&query_is_claimed(deps, address)?)?,
+    };
+    Ok(response)
+}
+
+#[allow(clippy::too_many_arguments)]
+fn execute_update_config(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    manager: Option<String>,
+    randdrop_denom: Option<String>,
+    nois_proxy_denom: Option<String>,
+    nois_proxy_amount: Option<Uint128>,
+    nois_proxy_address: Option<String>,
+) -> Result<Response, ContractError> {
+    let config = CONFIG.load(deps.storage)?;
+    // check the calling address is the authorised multisig
+    ensure_eq!(info.sender, config.manager, ContractError::Unauthorized);
+
+    let manager = match manager {
+        Some(ma) => deps.api.addr_validate(&ma)?,
+        None => config.manager,
+    };
+    let randdrop_denom = randdrop_denom.unwrap_or(config.randdrop_denom);
+    let nois_proxy_denom = nois_proxy_denom.unwrap_or(config.nois_proxy.price.denom);
+    let nois_proxy_amount = nois_proxy_amount.unwrap_or(config.nois_proxy.price.amount);
+    let nois_proxy_address = match nois_proxy_address {
+        Some(np) => deps.api.addr_validate(&np)?,
+        None => config.nois_proxy.address,
+    };
+
+    let nois_proxy = NoisProxy {
+        price: Coin {
+            denom: nois_proxy_denom,
+            amount: nois_proxy_amount,
+        },
+        address: nois_proxy_address,
+    };
+
+    CONFIG.save(
+        deps.storage,
+        &Config {
+            manager,
+            randdrop_denom,
+            nois_proxy,
+        },
+    )?;
+
+    Ok(Response::new().add_attribute("action", "update_config"))
+}
+
+// This function will call the proxy and ask for the randomness round
+pub fn execute_randdrop(
+    deps: DepsMut,
+    info: MessageInfo,
+    amount: Uint128,
+    proof: Vec<HexBinary>,
+) -> Result<Response, ContractError> {
+    let config = CONFIG.load(deps.storage)?;
+
+    if MERKLE_ROOT.may_load(deps.storage)?.is_none() {
```

Comment:
In cw-storage-plus 1.1.0 you can use `.has` on Item

Reply:
Oh, it's called `.exists`. See https://github.com/noislabs/nois-contracts/pull/259